monads/src/HW/Eval.hs

```diff
@@ -29,8 +30,51 @@ initialState = MachineState [] M.empty
 -- Execute a single instruction. 
 -- Successful evaluation does not produce any useful result: only the effect of modifying state matters. 
 execInstr :: (Ord v, Show v) => StackInstr v -> MyState (MachineState v) (Either (Error v) ())
-execInstr = undefined 
+execInstr (PushNum n) = do
+  MachineState stack env <- get
+  put $ MachineState (n:stack) env
+  return $ Right ()
```

Comment:
You don't modify `env`, so don't `get` it. Also, use `modify` instead of `put`. 
The comment is also valid for the code below.